cookbooks/php/recipes/fpm.rb

```diff
@@ -0,0 +1,163 @@
+class Chef::Recipe
+  include PhpHelpers
+end
+
+# Report to Cloud dashboard
+ey_cloud_report "processing php#{node["php"]["minor_version"]}" do
+  message "processing php - php-fpm #{node["php"]["minor_version"]}"
+end
+
+# Overwrite default php config
+directory "/etc/php/#{node["php"]["minor_version"]}/fpm" do
+  owner "root"
+  group "root"
+  recursive true
+  mode "0755"
+  action :create
+end
+
+cookbook_file "/etc/php/#{node["php"]["minor_version"]}/fpm/php.ini" do
+  source "php.ini"
+  owner "root"
+  group "root"
+  mode "0755"
+  backup 0
+end
+
+# create directory for fpm logs
+directory "/var/log/engineyard/php-fpm" do
+  owner node["owner_name"]
+  group node["owner_name"]
+  mode "0755"
+  action :create
+end
+
+# create error log for fpm
+file "/var/log/engineyard/php-fpm/error.log" do
+  owner node["owner_name"]
+  group node["owner_name"]
+  mode "0644"
+  action :create_if_missing
+end
+
+# create directory for unix socket(s)
+directory "/var/run/engineyard" do
+  owner node["owner_name"]
+  group node["owner_name"]
+  recursive true
+  mode "0755"
+  action :create
+end
+
+
+=begin
+bash 'eselect php and restart via monit' do
+  code <<-EOH
+    eselect php set fpm php#{node["php"]["minor_version"]}
+    EOH
+  not_if "php-fpm -v | grep PHP | grep #{node['php']['version']}"
+  notifies :run, 'execute[monit_restart_fpm]'
+end
+=end
+
+# get all applications with type PHP
+apps = node['dna']['applications'].select{ |app, data| data['recipes'].detect{ |r| r == 'php' } }
+# collect just the app names
+app_names = apps.collect{ |app, data| app }
+
+# generate global fpm config
+template "/etc/php-fpm.conf" do
+  owner node["owner_name"]
+  group node["owner_name"]
+  mode "0644"
+  source "fpm-global.conf.erb"
+  variables({
+    :apps => app_names
+  })
+#  notifies :restart, resources(:service => "php-fpm"), :delayed
+end
+
+# Can't access get_fpm_coount inside block
+app_fpm_count = (get_fpm_count / node['dna']['applications'].size)
+app_fpm_count = 1 unless app_fpm_count >= 1
+
+ssh_username = node.engineyard.environment.ssh_username
+# generate an fpm pool for each php app
+app_names.each do |app_name|
```

Comment:
No need, after https://github.com/engineyard/ey-cookbooks-stable-v6/pull/90/files is merged. Let's handle this on `env_vars` recipe for all cases/languages/webservers

Reply:
Pushed a commit that removed `cookbook_file "/data/#{app_name}/shared/config/env.custom"` and deleted files/default/env.custom.